twoliter/embedded/rpm2img

```diff
@@ -256,6 +257,48 @@ printf "%s\n" "${INVENTORY_DATA}" >"${ROOT_MOUNT}/usr/share/bottlerocket/applica
 # can access the inventory without needed to dig into the generated image.
 printf "%s\n" "${INVENTORY_DATA}" >"${OUTPUT_DIR}/application-inventory.json"
 
+# Install SBOM packages if directory provided
+SBOM_MOUNT=$(mktemp -d)
+trap 'rm -rf "${SBOM_MOUNT}"' EXIT
+
+if [[ -n "${SBOM_PACKAGE_DIR:-}" ]] && [[ -d "${SBOM_PACKAGE_DIR}" ]]; then
+  mapfile -t sbom_rpms < <(find "${SBOM_PACKAGE_DIR}" -maxdepth 1 -name "*.rpm" -type f)
+
+  if [[ ${#sbom_rpms[@]} -gt 0 ]]; then
+    mkdir -p "${SBOM_MOUNT}/var/lib/rpm"
+    rpm --root "${SBOM_MOUNT}" --initdb
+
+    CROSS_PREFIX=""
+    first_file=$(rpm -qpl "${sbom_rpms[0]}" 2>/dev/null | head -1)
+    if [[ "${first_file}" =~ ^(/[^/]+/sys-root)/ ]]; then
+      CROSS_PREFIX="${BASH_REMATCH[1]}"
+    fi
+
+    if [[ -n "${CROSS_PREFIX}" ]]; then
+      rpm -iv --ignorearch --badreloc --relocate "${CROSS_PREFIX}=/" --root "${SBOM_MOUNT}" "${sbom_rpms[@]}" >/dev/null 2>&1 ||:
+    else
+      rpm -iv --ignorearch --root "${SBOM_MOUNT}" "${sbom_rpms[@]}" >/dev/null 2>&1 ||:
+    fi
+  fi
+fi
+
+# Merge SBOMs into a single json file
+KIT_SBOMS_DIR="${SBOM_MOUNT}/usr/share/sboms"
+if [ -d "${KIT_SBOMS_DIR}" ]; then
+  IMAGE_SBOM_DIR="${ROOT_MOUNT}/usr/share/bottlerocket"
+  mkdir -p "${IMAGE_SBOM_DIR}"
+  for format in "spdx" "cyclonedx"; do
+    image_sbom="${format}-sbom.json"
+    image_sbom_path="${IMAGE_SBOM_DIR}/${image_sbom}"
+    find "${KIT_SBOMS_DIR}" -name "*-${format}.json" -type f -exec sbomtool merge --output "${image_sbom_path}" {} \+
+    if [[ -f "${image_sbom_path}" ]]; then
+      lz4 -z -f "${image_sbom_path}" "${image_sbom_path}.lz4"
```

Comment:
Let's move this after the "copy out", and condition the compression on whether or not we're using EROFS.